src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py

```diff
@@ -619,6 +638,8 @@ def __call__(
         ] = None,
         strength: float = 0.3,
         num_inference_steps: int = 50,
+        max_inference_steps: Optional[int] = None,
```

Comment:
```suggestion
        final_denoising_step: Optional[int] = None,
```

Reply:
so if you use percentages the accuracy of the calculation and converting to integer means you might misalign timesteps, which results in artifacting

Reply:
Fair point!

Reply:
Actually I think this can't really happen if we always round, e.g.:

```py
max_inference_steps = int(round(denoising_end * num_inference_steps))
first_inference_steps = int(round(denoising_start * num_inference_Steps))
```

`denoising_end` should always be equal to `1 - denoising_start` and in this case we will always have:

```py
num_inference_steps = max_inference_steps + first_inference_steps
```

no?

Reply:
it happens when you have precise values for `strength` that result in remainders in the `int()` casting that become discarded. i have tested it:
![1689000316 7576849ca8f41e6b56b1ff29f093680e55fbc6](https://github.com/huggingface/diffusers/assets/59658056/727c7404-8103-4b3c-866a-91af6d57791a)

In this test, `num_inference_steps` is set to `27` and `strength` is set to `0.73` - the calculation says that the model should run for about `19.71` steps, but since we can't run a fractional step, this would be truncated by `int()` to `19`, but the value for `begin_inference_step` is `7.27` which truncates to `7`.

This results in a misalignment of one step, which is, apparently, enough to disrupt the output altogether.

We can use round(), but instead, I have opted for what I've considered a more robust approach.

Furthermore, there are some prompts/seeds/subject material where these artifacts prove useful in making the output more variable and creative. Having the option to set those and override our best judgement seems like one of the things that makes Diffusers great, and I wanted to retain that.

Reply:
+1 on @patrickvonplaten suggestion on using a percentage 

It seems to me we will just have to set  `denoising_end = denoising_start = 1- strength`? 

then,
`final_inference_steps = int(denoising_start * num_inference_steps)`
`begin_inference_steps = int(denoising_end * num_inference_steps)`

so in the example `num_inference_step = 27`, `strength= 0.73`,  we will have `final_inference_steps = begin_inference_steps = 7`, i.e  run `7` steps in base model and `20` in refiner - is this what's intended?

Reply:
ive tested this, and while it seems like it should work, it has really broken the results for me, overall. i am digging into it.